Epochal data import

cawasser opened the pull request:
this is kind of complicated:

1. implementation of a "materialized view" that accumulated the deltas described by the Demo CONUS.xslx file (things start, things end)
2. take the completed MV and put it into Datascript (a different conn than the one in db.excel-data)
3. originally built a set of queries using the MV directly, and then convert to using Datascript again

This PR includes updates from the ww-layers branch.

Commit:
not working, in progress

Commit:
Merge branch 'master' into resizing

Commit:
widgets all size correctly!

Commit:
cleanup

Commit:
fixed timeline with a css tweak.  carousel not working

Commit:
Highcharts in carousel working!

Commit:
carousel tweaks

Commit:
fixed data table and health/status table sizing

cawasser committed:
- working (refresh-sources...) as a means of working at the REPL and getting the data sources to provide updated information without restarting the server

cawasser committed:
- (add-subscribers...) fixed
- (remove-subscriber...) updated
- added POST /logout to client when clicking the "Logout" button ot notify the server that the user has logged out (also cleaned up when the websocket closes)

cawasser committed:
- added repl call to refresh :current-time for testing and example purposes

cawasser committed:
- adding documentation for developing the server in the REPL.

Commit:
changed add-widget button to add carousel, locked size

Commit:
merged with master

Commit:
removed carousel service from modal list

Commit:
cleanup

cawasser committed:
- return multiple series in both :sankey-service and :signal-path-service

cawasser committed:
- present 2 different way of showing multi-sankey data: as multiple series on a standard sankey Highcharts widget, or as separate "slides" in a carousel
- add another button of create a "sankey" flavored carousel (temporary experimentation)

cawasser committed:
- changes signal-path-serivce to :grouped-from-to-n which only maps to :sankey-carousel-widget
- removed now extraneous "Add Sankey" button from the home-page
- removed dummy "carousel-service" from the database

cawasser committed:
- fixed highcharts dependencies

Commit:
updated db to remove carousel service.  cleaned up comments

Commit:
comment comments

cawasser committed:
- new documentation for "How Widgets Work"

Commit:
heatmap comment cleanup

cawasser committed:
- document how widgets are structured (what contains what)

cawasser committed:
Merge branch 'multi-sankey' into 9102-demo

# Conflicts:
#	src/cljs/vanilla/add_widget.cljs
#	src/cljs/vanilla/widgets/carousel.cljs
#	src/cljs/vanilla/widgets/make_chart.cljs

cawasser committed:
- lots of experimentation (too much, in fact) with the new "event stream" interop format, prepping for a trim

Commit:
broken, but globe and navbar show up and render

cawasser committed:
- close to a basic materialized view of the event stream

cawasser committed:
Merge branch 'repl-the-server' into 9102-demo

# Conflicts:
#	README.md
#	src/clj/vanilla/terminal_list_service.clj

cawasser committed:
- new implementation of signal-path-service, task-service, and terminal-location-service
- minor updates to carousel and timeline widgets

cawasser committed:
- send task-service dates to client as #inst

cawasser committed:
- fix :end to pull the correct data and get the "last" (latest) date in the set

cawasser committed:
- switch back to "year" zoom-level for timeline widget. seems to be a bug with the other levels

cawasser committed:
- beam-location-service now using (mv/ka-beam-query)

cawasser committed:
- use Demo.xslx

cawasser committed:
- corrected for satellite names in the example data

cawasser committed:
- corrected for data in the example

cawasser committed:
- cleanup rich comment data
- support separate beam services form a single import sheet

cawasser committed:
- tweaks to match sample Excel data

cawasser committed:
- more cities around the world
- several starting points for centering the map

cawasser committed:
- cleanup services to match new Demo.xlsx format

cawasser committed:
- rearrange layers to get more consistent rendering form WorldWind

cawasser committed:
- UI cleanup for better presentation and clarity

cawasser committed:
- cleanup materialized_view.clj for clarity and presentation

cawasser committed:
- data-source cleanup for better presentation and clarity

cawasser committed:
- UI cleanup for better presentation and clarity

cawasser committed:
- UI cleanup for better presentation and clarity

cawasser committed:
- updated to CONUS demo data
- upgraded shadow-cljs to 2.10.9 (need to update highcharts, react-pure-grid as well)

cawasser committed:
- try turning the cities layer back on

cawasser committed:
- added current-user tag to the UI

cawasser committed:
- added :current-user to the main UI following login
- added :inactive to the stop-light widget for better presentation

Commit:
IT WORKS!!!! layers show and toggle

Commit:
fixed subscribe overload issue

Commit:
styling tweaks and icon attempt

Commit:
more styling tweaks

Commit:
changed ww-react-globe.js to handle options for layers

Commit:
pulled the bootstrap css and js in

Commit:
css tweak

cawasser committed:
Merge branch 'wwlayers' into epochal-data-import

# Conflicts:
#	src/cljs/vanilla/mapping/layer_management.cljs
#	src/cljs/vanilla/widgets/worldwind.cljs
#	vanilla_default

cawasser committed:
- use map for adding layers to WW

cawasser committed:
Merge branch 'wwlayers' into epochal-data-import

# Conflicts:
#	src/cljs/vanilla/mapping/layer_management.cljs
#	src/cljs/vanilla/widgets/worldwind.cljs

- setting up for combining terminal and beam data into a single layer, one for each epoch

cawasser committed:
- WIP: combining terminal and beam data into a single layer, one for each epoch (not working yet)

Commit:
fixed disappearing navbar and overlay sizing

Commit:
refactored data-grid into widget defs for better flexibility

cawasser committed:
- refactor to combine signal and beams into a single Materialized View over time (epochs)

cawasser committed:
Merge branch 'epochal-data-import' of https://github.com/cawasser/vanilla into epochal-data-import

cawasser committed:
- fixes for terminal-location - WORKING!

cawasser committed:
- fixes for beam-location - WORKING!

Commit:
icons working, minor cleanup

cawasser committed:
- removed title from the "nav bar" for the WW map (cleaner UI)
- after moving (lm/make-layers) out of the :render phase, we can move the data-source subscriptions BACK to lm and keep the WW map a little more generic

cawasser committed:
- let's start moving back to datascript!
- partial implementation on datascript (terminals and beams)
- should we use a macro to streamline the query data munging?

cawasser committed:
- developed a few helper functions to make doing the datascript querying a little easier, so you don't forget to group, sort, and merge all the data

cawasser committed:
- reworked signal-path and mission queries to use datascript and the new Materialized View

cawasser committed:
- minor clean-up and some comparisons with the (mv/xxx-query) series of implementations. only difference is that the "empty" last epoch is NOT included in the datascript implementation (which makes perfect sense)
- also an issue with the bem results NOT including the first 3 epoch because they are missing inf the source data (to be fixed)

cawasser committed:
- datascript implementation migrated to final namespaces

cawasser committed:
- cleanup MV implementation (remove duplicate/unused code)
- init MV during server startup
- add minor configuration options for Sankey diagram to signal-path-service

cawasser committed:
- updated beams for SAT1 to start ain the first epoch
- include :satellite-id as part of beam-data

cawasser committed:
- documentation updates
- in preparation for tagging as v0.5.0

cawasser committed:
- documentation updates

cawasser committed:
- documentation updates

cawasser committed:
- added discussion of `query-thread` adn some examples of use in the project

cawasser committed:
- docstring updates

cawasser committed:
Merge branch 'master' into epochal-data-import

# Conflicts:
#	package.json
#	src/cljs/vanilla/add_widget.cljs
#	src/cljs/vanilla/widgets/carousel.cljs
#	src/cljs/vanilla/widgets/make_chart.cljs
#	vanilla_default

austinpaine reviewed (approved):
Everything looks good to me!  Good documentation and the subscription manager changes will be helpful going forward

Commit:
remove unused list icon png

cawasser committed:
- added an "epoch-layer" to show a label for which epoch(s) are visible
- refactored the "starting-locations" out of WW so we can use the data in make-layers
- updated the beam IDs to be strings in the CONUS data

cawasser committed:
Merge branch 'epochal-data-import' of https://github.com/cawasser/vanilla into epochal-data-import

Commit:
css tweaks for login modal and card

Commit:
css fix for close button of layer popup, added new base layer

cawasser committed:
- corrected "finally" condition in excel-data
- corrected path to resource file (so it works in Docker)
- refactored (get-all-events) to open the excel file only once in a try/catch block

gpierce3 reviewed (approved):
built/ran in dev, uber, docker.  Everything worked as expected.
Really looking good!

I'm approving, but two questions:

Are we not retaining layer selection data for WW when logging off/on in this release?

Looks like we reference 0.5.0 release in readme.md, but snapshot on web page is still 0.4.2-blah.
Is this what we want?

cawasser commented:
Currently we aren't saving ANY of the user selections on widgets. I added a new task in JIRA.

I plan on bumping the version to 0.5.0 since this adds significant new features that the user will see, but the bump actually happens "after" the merge, because of how git tags work

cawasser committed:
- updated Demo CONUS.xslx to better match beams to terminal locations
